source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -255,6 +255,40 @@ def sort_journeys(resp, journey_order, clockwise):
         resp.journeys.sort(key=cmp_to_key(journey_sorter[journey_order](clockwise=clockwise)))
 
 
+def fill_missing_co2_emission(pb_resp, api_request, instance, request_id):
+    if not pb_resp.journeys:
+        return
+
+    car_mode = None
+    for j in pb_resp.journeys:
+        if 'car' in j.tags:
+            for s in j.sections:
+                if (s.type == response_pb2.STREET_NETWORK and s.street_network.mode == response_pb2.Car
+                 and not s.co2_emission.value):
+                    if not car_mode:
+                        req = request_pb2.Request()
+                        req.requested_api = type_pb2.PTREFERENTIAL
+                        req.ptref.requested_type = type_pb2.PHYSICAL_MODE
+                        req.ptref.filter = 'physical_mode.uri={uri}'.format(uri='physical_mode:Car')
+                        req.ptref.start_page = 0
+                        req.ptref.count = 1
+                        req.ptref.depth = 1
+
+                        car_mode = instance.send_and_receive(req, request_id=request_id)
+
+                    if not (car_mode.physical_modes
+                            and car_mode.physical_modes[0].HasField('co2_emission')
+                            and car_mode.physical_modes[0].co2_emission.value):
+                        return
+
+                    distance = 0
+                    for p in s.street_network.path_items:
+                        distance += p.length
```

Comment:
Don't we already have the total distance somewhere ? Like in `s.street_network.length` ? 
Since path_items may be a big list, I am a bit worried about time it will take to make this computation

Reply:
Yes my bad I looked for this value but I missed it.